### ISSUE TITLE
SLIP-0013 worked example incorrect

### ISSUE THREAD

**Issue by saleemrashid** [satoshilabs/slips#108]
The unhardened indexes in the [worked example](https://github.com/satoshilabs/slips/blob/master/slip-0013.md#worked-example) are incorrect, as far as I can tell.

I have attached two implementations of the worked example. The output is:

```
A  = 490267344
A' = 2637750992
B  = 697598796
B' = 2845082444
C  = 1613620211
C' = 3761103859
D  = 1858012177
D' = 4005495825
```

However, according to the worked example, the expected output is:

```diff
@@ -1,8 +1,8 @@
-A  = 490267344
+A  = 221831888
 A' = 2637750992
-B  = 697598796
+B  = 160727884
 B' = 2845082444
-C  = 1613620211
+C  = 3007475
 C' = 3761103859
-D  = 1858012177
+D  = 247399441
 D' = 4005495825
```

/cc @gary-rowe
/cc @joaojoyce #107 

```c
/* Little-endian only */

#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <string.h>

#include "sha2.h"

int main(int argc, char **argv) {
    (void) argc;
    (void) argv;

    uint32_t index = 0;
    const char *uri = "https://satoshi@bitcoin.org/login";

    SHA256_CTX ctx;
    sha256_Init(&ctx);

    sha256_Update(&ctx, (const uint8_t *) &index, sizeof(index));
    sha256_Update(&ctx, (const uint8_t *) uri, strlen(uri));

    uint8_t hash[SHA256_DIGEST_LENGTH];
    sha256_Final(&ctx, hash);

    const uint32_t *fingerprint = (const uint32_t *) hash;
    for (size_t i = 0; i < 4; i++) {
        uint32_t index = fingerprint[i];

        char name = 'A' + i;
        printf("%c  = %"PRIu32"\n", name, index & ~0x80000000);
        printf("%c' = %"PRIu32"\n", name, index |  0x80000000);
    }
}
```

```python
from __future__ import print_function
import hashlib, string, struct

index = 0
uri = b"https://satoshi@bitcoin.org/login"

h = hashlib.sha256()
h.update(struct.pack("<L", index))
h.update(uri)
hash = h.digest()

fingerprint = struct.unpack_from("<4L", hash)
for name, index in zip(string.ascii_uppercase, fingerprint):
	print("{}  = {}".format(name, index & ~0x80000000))
	print("{}' = {}".format(name, index |  0x80000000))
```

**Comment by saleemrashid** [comment]
PHP implementation:

```php
<?php
$index = 0;
$uri = "https://satoshi@bitcoin.org/login";

$hash = hash("sha256", pack("V", $index) . $uri, TRUE);

$fingerprint = unpack("VA/VB/VC/VD", $hash);
foreach ($fingerprint as $name => $index) {
    printf("%s  = %u\n", $name, $index & ~0x80000000);
    printf("%s' = %u\n", $name, $index |  0x80000000);
}
?>
```

**Comment by prusnak** [comment]
Fixed in 4131f572653db15463ced4fa702485da37ea6076
